src/Utilities/Compiler/PooledObjects/PooledDictionary.cs

```diff
@@ -41,6 +41,24 @@ private PooledDictionary(ObjectPool<PooledDictionary<K, V>>? pool, IEqualityComp
             return result;
         }
 
+        public ImmutableDictionary<TKey, TValue> ToImmutableDictionaryAndFree<TKey, TValue>(
```

Comment:
I have kept the 2 full implementations but I could refactor the parameterless method to call this one.